File: 1.INTRODUCCION-POO/EJERCICIO-9/JAVA/Computadora.java
import java.util.ArrayList;
import java.util.List;


class Computadora {
    private boolean estado;
    private List<Componente> componentes;

    public Computadora() {
        this.estado = false;
        this.componentes = new ArrayList<>();
    }

    public void encender() {
        if (!estado) {
            estado = true;
            System.out.println("La computadora está encendida.");
        } else {
            System.out.println("La computadora ya está encendida.");
        }
    }

    public void apagar() {
        if (estado) {
            estado = false;
            System.out.println("La computadora está apagada.");
        } else {
            System.out.println("La computadora ya está apagada.");
        }
    }

    public void mostrarEstado() {
        if (estado) {
            System.out.println("La computadora está encendida.");
        } else {
            System.out.println("La computadora está apagada.");
        }
    }

    public void agregarComponente(Componente componente) {
        componentes.add(componente);
        System.out.println("Componente " + componente.getNombre() + " agregado.");
    }

}

File: 1.INTRODUCCION-POO/EJERCICIO-9/PYTHON/Computadora.py
class Componente:
    def __init__(self, nombre):
        self.nombre = nombre

    def get_nombre(self):
        return self.nombre

class Computadora:
    def __init__(self):
        self.estado = False
        self.componentes = []

    def encender(self):
        if not self.estado:
            self.estado = True
            print("La computadora está encendida.")
        else:
            print("La computadora ya está encendida.")

    def apagar(self):
        if self.estado:
            self.estado = False
            print("La computadora está apagada.")
        else:
            print("La computadora ya está apagada.")

    def mostrar_estado(self):
        if self.estado:
            print("La computadora está encendida.")
        else:
            print("La computadora está apagada.")

    def agregar_componente(self, componente):
        self.componentes.append(componente)
        print(f"Componente {componente.get_nombre()} agregado.")

def main():
    mi_computadora = Computadora()

    mi_computadora.agregar_componente(Componente("CPU"))
    mi_computadora.agregar_componente(Componente("RAM"))
    mi_computadora.agregar_componente(Componente("Disco Duro"))

    mi_computadora.mostrar_estado()
    mi_computadora.encender()
    mi_computadora.mostrar_estado()
    mi_computadora.apagar()
    mi_computadora.mostrar_estado()

if __name__ == "__main__":
    main()
